src/Microsoft.IdentityModel.Logging/IdentityModelTelemetryUtil.cs

```diff
@@ -0,0 +1,139 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Net.Http;
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// Provides a way to add and remove telemetry data.
+    /// </summary>
+    public static class IdentityModelTelemetryUtil
+    {
+        internal const string skuTelemetry = "x-client-SKU";
+        internal const string versionTelemetry = "x-client-Ver";
+        internal static readonly List<string> defaultTelemetryValues = new List<string> { skuTelemetry, versionTelemetry };
+        internal static readonly ConcurrentDictionary<string, string> telemetryData = new ConcurrentDictionary<string, string>()
+        {
+            [skuTelemetry] = ClientSku,
+            [versionTelemetry] = ClientVer
+        };
+
+        /// <summary>
+        /// Get the string that represents the client SKU.
+        /// </summary>
+        public static string ClientSku =>
+#if NET45
+            "ID_NET45";
+#elif NET461
+            "ID_NET461";
+#elif NET472
+            "ID_NET472";
+#elif NETSTANDARD2_0
+            "ID_NETSTANDARD2_0";
+#endif
+
+        /// <summary>
+        /// Get the string that represents the client version.
+        /// </summary>
+        public static string ClientVer => typeof(IdentityModelTelemetryUtil).GetTypeInfo().Assembly.GetName().Version.ToString();
+
+        /// <summary>
+        /// Adds a key and its value to the collection of telemetry data.
+        /// </summary>
+        /// <param name="key"> The name of the telemetry.</param>
+        /// <param name="value"> The value of the telemetry.</param>
+        /// <returns> true if the key is successfully added; otherwise, false.</returns>
+        public static bool AddTelemetryData(string key, string value)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                LogHelper.LogArgumentNullException(nameof(key));
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(value))
+            {
+                LogHelper.LogArgumentNullException(nameof(value));
+                return false;
+            }
+
+            if (defaultTelemetryValues.Contains(key))
+            {
+                LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.MIML10003, defaultTelemetryValues)));
+                return false;
+            }
+
+            telemetryData[key] = value;
+            return true;
+        }
+
+        /// <summary>
+        /// Removes a key and its value from the collection of telemetry data.
+        /// </summary>
+        /// <param name="key"> The name of the telemetry.</param>
+        public static void RemoveTelemetryData(string key)
```

Comment:
could also return a bool. Throwing seems aggressive for telemetry data, imo.

Reply:
Fixed it.